common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -25,6 +25,8 @@
 import com.google.common.collect.Lists;
 import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.util.DiskChecker;
```

Comment:
In our case, we have about 200 nodes which have the first disk broken, and spark application is not minority. We can't accept lost so much nodes just because the first disk was broken, also we can't accept let spark application fail randomly (I already patched 1.6 use same logic in our cluster so we can keep enable spark dynamic allocation).

If no one see any value of the improvement, that's fine, I'll close this PR.
Thank you for your time.